node/pkg/common/ethish.go

```diff
@@ -0,0 +1,31 @@
+// This specifies the interface to the chain specific Eth / EVM libraries.
+// This interface should be implemented for each chain that has a unique go-ethereum or "go-ethereum-ish" library.
+
+package common
+
+import (
+	"context"
+
+	ethereum "github.com/ethereum/go-ethereum"
+	ethCommon "github.com/ethereum/go-ethereum/common"
+	ethTypes "github.com/ethereum/go-ethereum/core/types"
+	ethEvent "github.com/ethereum/go-ethereum/event"
+
+	ethAbi "github.com/certusone/wormhole/node/pkg/ethereum/abi"
+
+	"go.uber.org/zap"
+)
+
+type Ethish interface {
+	SetLogger(l *zap.Logger)
+	DialContext(ctx context.Context, rawurl string) error
+	NewAbiFilterer(address ethCommon.Address) error
+	NewAbiCaller(address ethCommon.Address) error
+	GetCurrentGuardianSetIndex(ctx context.Context) (uint32, error)
```

Comment:
What's the point of having three separate "init" methods?

Reply:
This is to maintain the full flexibility of the go-ethereum library. For instance, parse_eth_tx uses this, but it does not need the caller, so it only calls the first two init methods. Going to leave this as is.